### PR TITLE
[Do not merge] Try to figure out allow_paths for new solc versions

### PR DESCRIPTION
**Pull request by miohtama**
**Deleting and rebasing on master**

**Work in progress - do not merge - opened only for the discussion**

### What was wrong?

Populus cannot pass external paths (lib sources) to new versions of solc correctly. The last working version was solc 0.4.8 and Populus 1.6.9. This prevents using Zeppelin contracts inside your own contract repository.

### How was it fixed?

* Need to pass `allow_paths` on `solc` command line. pysolc support is there, but populus did not handle command line

* (Potentially) need to pass some other settings in Solidity JSON input description `settings`

* Split config file compiler section so that we can specify both 1) JSON input description `settings` 2) command line argument

An example config section:

    "backends": {
      "SolcAutoBackend": {
        "class": "populus.compilation.backends.SolcAutoBackend",
        "settings": {
          "solidity_input_description": {
            "optimizer": {
              "enabled": true,
              "runs": 500
            },
            "outputSelection": {
              "*": {
                "*": [
                  "abi",
                  "metadata",
                  "evm.bytecode",
                  "evm.deployedBytecode"
                ]
              }
            }
          },
          "command_line_options": {
            "allow_paths": "/"
          }
        }
      },

#### Cute Animal Picture

```
 
                         (__)                        (__)
                 ^^      (oo)                        (--)
             ^^^^ /-------\/                        /-\/-\
          ^^^^^  / |     ||                        /|    |\
        ^^^^^   *  ||----||                       ^ |    | ^
     ^^^^^^^^  ====^^====^^====                     |    |
 ^^^^^^^^^^^^^/                                     /----\
 ^^^^^^^^^^^^^^^^^^                                /    \ \
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^               ^      * ^
      Cow Hanging Ten at Malibu           Cow sunning at Fort Lauderdale
                                              (What a bod, huh guys?)
```

### CODE REVIEW

**Comment by miohtama** [comment]
The question is what are these errors and how do I get rid of them:

```
Command line settings are {'allow_paths': '/'}
['solc', '--allow-paths', '/', '--standard-json']
Traceback (most recent call last):
  File "/Users/mikko/code/wattcoin/ico/venv/bin/populus", line 11, in <module>
    load_entry_point('populus', 'console_scripts', 'populus')()
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/mikko/code/wattcoin/ico/venv/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/mikko/code/wattcoin/populus/populus/cli/compile_cmd.py", line 40, in compile_cmd
    _, compiled_contracts = compile_project_contracts(project)
  File "/Users/mikko/code/wattcoin/populus/populus/compilation/__init__.py", line 53, in compile_project_contracts
    validate_compiled_contracts(compiled_contracts)
  File "/Users/mikko/code/wattcoin/populus/populus/utils/compile.py", line 208, in validate_compiled_contracts
    raise ValidationError(error_message)
populus.exceptions.ValidationError: The following compiled contracts did not pass validation:
--------MultiSigWallet-----------
{'payable': True, 'type': 'fallback'} is not valid under any of the given schemas
--------TimeVault-----------
{'payable': False, 'type': 'fallback'} is not valid under any of the given schemas
--------RelaunchedCrowdsale-----------
{'payable': True, 'type': 'fallback'} is not valid under any of the given schemas

```

**Comment by pipermerriam** [comment]
Those errors are from the json-schema which is used to validate the contract data.  I opened up an independent issue for it: https://github.com/pipermerriam/populus/issues/328

File: populus/compilation/backends/solc_standard_json.py [hunk 1]
```diff
@@ -114,10 +114,15 @@ def get_compiled_contracts(self, source_file_paths, import_remappings):
                 'remappings': import_remappings
             }
         }
-        std_input['settings'].update(self.compiler_settings)
 
+        std_input['settings'].update(self.compiler_settings.get("solidity_input_description", {}))
```
Comment (pipermerriam) [inline populus/compilation/backends/solc_standard_json.py]: I think that calling this `"standard_input"` is a good name/key for this setting.

Comment (miohtama) [inline populus/compilation/backends/solc_standard_json.py]: Ok, will rename.

**Pull request closed without merging**